### PR TITLE
Longhorn Manual Resolution

### PR DESCRIPTION
**Pull request by karl-cardenas-coding**
## Describe the Change

This PR adds instructions that address the issues pertaining to the ShareManger remaining in the incorrect state when nodes redeploy. This pertains to RWX volume.

## Review Changes

💻 [Preview URL](https://659719f2a29484372a7f56ac--docs-spectrocloud.netlify.app/integrations/longhorn#deadlock-with-rwx-mode)

🎫 [DOC-990](https://spectrocloud.atlassian.net/browse/DOC-990)

### CODE REVIEW

File: docs/docs-content/integrations/longhorn.md [hunk 1]
```diff
@@ -17,10 +17,74 @@ Longhorn creates a dedicated storage controller for each block device volume and
...
+
+The table lists commonly used parameters you can configure when adding this pack.
+
+| Parameter                | Description                                            | Default                                     |
+|-------------------------|--------------------------------------------------------|---------------------------------------------|
+| `namespace`               | The namespace where the Longhorn will be installed. | `longhorn-system` |
+| `defaultClass`            | The volume type to be used. | `true` |
+| `defaultFsType`           | The default file system.    | `ext4` |
+| `defaultClassReplicaCount`| The default number of copies of data store in your cluster.   | `3`         |
+| `defaultDataLocality`     | The default location where data computation will occur. | `disabled` Best effort |
```
Comment (lennessyy) [inline docs/docs-content/integrations/longhorn.md:37]: This is a little confusing to me. If it's a location, why is the default "disabled"? And maybe we can expand on the "best effort" a bit?

File: docs/docs-content/integrations/longhorn.md [hunk 2]
```diff
@@ -17,10 +17,74 @@ Longhorn creates a dedicated storage controller for each block device volume and
...
+| `reclaimPolicy`           | This means that a dynamically provisioned volume will be automatically deleted when deletes when corresponding PersistentVolumeClaim is deleted. For important data, it is more appropriate to use the "Retain" policy | `Delete`  |
```
Comment (lennessyy) [inline docs/docs-content/integrations/longhorn.md:38]:
```suggestion
| `reclaimPolicy`           | The policy that regulates dynamically provisioned volumes when its corresponding PersistentVolumeClaim is deleted. The default policy is `Delete`. This means that a dynamically provisioned volume will be automatically deleted when corresponding PersistentVolumeClaim is deleted. For important data, it is more appropriate to use the "Retain" policy. | `Delete`  |
```

File: docs/docs-content/integrations/longhorn.md [hunk 3]
```diff
@@ -17,10 +17,74 @@ Longhorn creates a dedicated storage controller for each block device volume and
...
+To resolve the deadlock, you must manually update the status of the Longhorn Storage Manager. Use the following command to resolve the deadlock. Replace `PVC-ID` with the ID of the Persistent Volume Claim (PVC) that is in the `longhorn-system` namespace that and in **RWX** mode.
+
+```bash
+kubectl --namespace longhorn-system patch lhsm PVC-ID \
+--type=merge --subresource status --patch 'status: {state: error}'
+```
+
+:::tip
+
+To identify the ID of the PVC, use the following command to list all PVCs in the `longhorn-system` namespace. Seach for the PVC that has the **ACCESS MODE** set to **RWX**.
```
Comment (lennessyy) [inline docs/docs-content/integrations/longhorn.md:71]:
```suggestion
To identify the ID of the PVC, use the following command to list all PVCs in the `longhorn-system` namespace. Search for the PVC that has the **ACCESS MODE** set to **RWX**.
```

File: docs/docs-content/integrations/longhorn.md [hunk 4]
```diff
@@ -17,10 +17,74 @@ Longhorn creates a dedicated storage controller for each block device volume and
...
+For more information, check out Longhorn guide on [How to Create Volumes](https://longhorn.io/docs/1.4.0/volumes-and-nodes/create-volumes/).
+
+
+## Troubleshooting
+
+Below are some common issues you may encounter when using Longhorn. Use the following information to help you troubleshoot and resolve the issue.
+
+### Deadlock With RWX Mode
+
+Starting with version 1.5.X, a known issue exists where a deadlock may occurs when using a RWX volume and a cluster node is recreated. This issue is disclosed as a known issue in the [Longhorn GitHub repository](https://github.com/longhorn/longhorn/wiki/Release-Known-Issues#v153).
```
Comment (lennessyy) [inline docs/docs-content/integrations/longhorn.md:60]:
```suggestion
Starting with version 1.5.X, a known issue exists where a deadlock may occur when using a RWX volume and a cluster node is recreated. This issue is disclosed as a known issue in the [Longhorn GitHub repository](https://github.com/longhorn/longhorn/wiki/Release-Known-Issues#v153).   
```

File: docs/docs-content/integrations/longhorn.md [hunk 5]
```diff
@@ -17,10 +17,74 @@ Longhorn creates a dedicated storage controller for each block device volume and
...
+
+| Parameter                | Description                                            | Default                                     |
+|-------------------------|--------------------------------------------------------|---------------------------------------------|
+| `namespace`               | The namespace where the Longhorn will be installed. | `longhorn-system` |
+| `defaultClass`            | The volume type to be used. | `true` |
+| `defaultFsType`           | The default file system.    | `ext4` |
+| `defaultClassReplicaCount`| The default number of copies of data store in your cluster.   | `3`         |
+| `defaultDataLocality`     | The default location where data computation will occur. | `disabled` Best effort |
+| `reclaimPolicy`           | This means that a dynamically provisioned volume will be automatically deleted when deletes when corresponding PersistentVolumeClaim is deleted. For important data, it is more appropriate to use the "Retain" policy | `Delete`  |
+| `migratable`              | The ability to transfer data to another data storage systems | `false`   |
```
Comment (lennessyy) [inline docs/docs-content/integrations/longhorn.md:39]:
```suggestion
| `migratable`              | The ability to transfer data to another data storage systems. | `false`   |
```

**Review by lennessyy — APPROVED** [review]
Approved with some minor comments

File: docs/docs-content/integrations/longhorn.md (hunk 1 above)
Comment (karl-cardenas-coding) [reply docs/docs-content/integrations/longhorn.md:37]: Agree, I'm gonna also link to their parameter page.

File: docs/docs-content/integrations/longhorn.md [hunk 6]
```diff
@@ -17,38 +17,110 @@ Longhorn creates a dedicated storage controller for each block device volume and
...
+| `charts.longhorn.global.persistence.defaultDataLocality`     | The default location where data computation will occur. | `disabled`|
+| `charts.longhorn.global.persistence.reclaimPolicy`           | The policy that regulates dynamically provisioned volumes when its corresponding PersistentVolumeClaim is deleted. The default policy is `Delete`. This means that a dynamically provisioned volume will be automatically deleted when corresponding PersistentVolumeClaim is deleted. For important data, it is more appropriate to use the "Retain" policy. | `Delete`  |
+| `charts.longhorn.global.persistence.migratable`              | The ability to transfer data to another data storage systems. | `false`   |
+| `charts.longhorn.global.persistence.recurringJobSelector.enable`   | The management of recurring jobs. You can enable this feature and type a comma-separated list of jobs to activate.  | `false`   |
+
+## Usage
+
+The official Longhorn documentation contains several guides and examples. The following guides are a good place to start when learning how to use Longhorn.
+
+- [How to Create Volumes](https://longhorn.io/docs/1.5.3/volumes-and-nodes/create-volumes/).
```
Comment (lennessyy) [inline docs/docs-content/integrations/longhorn.md:53]:
```suggestion
- [How to Create Volumes](https://longhorn.io/docs/1.5.3/volumes-and-nodes/create-volumes/)
```

**Review by lennessyy — APPROVED** [review]
The refactor looks great. I also added another known issue to the list from PE-3340.

**Pull request merged**